scripts/apps/fields/custom-fields-provider.ts

```diff
@@ -0,0 +1,33 @@
+import {IArticle} from 'superdesk-interfaces/Article';
+
+export interface IEditorComponentProps {
+    item: IArticle;
+    value: any;
+    setValue: (value: any) => void;
+    readOnly: boolean;
+}
+
+export interface IPreviewComponentProps {
+    item: IArticle;
+    value: any;
+}
+
+interface IFieldOptions {
+    label: string;
+    editorComponent: React.ComponentClass<IEditorComponentProps>;
+    previewComponent: React.ComponentClass<IPreviewComponentProps>;
+}
+
+export class CustomFields {
+    fields: {[key: string]: IFieldOptions};
+
+    constructor() {
+        this.fields = {};
+    }
+
+    add(fieldId: string, options: IFieldOptions) {
+        this.fields[fieldId] = options;
+    }
+}
+
+export const fields = new CustomFields();
```

Comment:
Accessing fields looks ugly `fields.fields`. What about getting rid of the class?

```
export const fields: fields: {[key: string]: IFieldOptions} = {};
export const addField = (fieldId: string, options: IFieldOptions) => {
    fields[fieldId] = options;
}
```